examples/gno.land/r/gov/dao/proxy/proxy.gno

```diff
@@ -0,0 +1,141 @@
+package proxy
+
+import "std"
+
+// dao is the actual govDAO implementation, having all the needed business logic
+var dao DAO
+
+// AllowedDAOs contains realms that can be used to update the actual govDAO implementation,
+// and validate Proposals.
+// This is like that to be able to rollback using a previous govDAO implementation in case
+// the latest implementation has a breaking bug. After a test period, a proposal can be
+// executed to remove all previous govDAOs implementations and leave the last one.
+var AllowedDAOs []string
+
+// proposals contains all the proposals in history.
+var proposals []*Proposal
+
+func init() {
+	AllowedDAOs = []string{"gno.land/r/gov/dao/v3/impl"}
+}
+
+// Render calls directly to Render's DAO implementation.
+// This allows to have this realm as the main entry point for everything.
+func Render(p string) string {
+	return dao.Render(p)
+}
+
+// MustCreateProposal is an utility method that does the same as CreateProposal,
+// but instead of erroing if something happens, it panics.
+func MustCreateProposal(r ProposalRequest) ProposalID {
+	pid, err := CreateProposal(r)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return pid
+}
+
+// CreateProposal will try to create a new proposal, that will be validated by the actual
+// govDAO implementation. If the proposal cannot be created, an error will be returned.
+func CreateProposal(r ProposalRequest) (ProposalID, error) {
+	//TODO: get prev realm here and remove the author coming from the govDAO impl
+	author, err := dao.PreCreateProposal(r)
+	if err != nil {
+		return -1, err
+	}
+
+	p := &Proposal{
+		Author:      author,
+		Title:       r.Title,
+		Description: r.Description,
+		Executor:    newExecutor(r.Callback, r.Metadata),
+		AllowedDAOs: AllowedDAOs,
+	}
+
+	pid := ProposalID(len(proposals))
+
+	proposals = append(proposals, p)
+
+	dao.PostCreateProposal(r, pid)
+
+	return pid, nil
+}
+
+func MustVoteOnProposal(r VoteRequest) {
+	if err := VoteOnProposal(r); err != nil {
+		panic(err.Error())
+	}
+}
+
+// VoteOnProposal sends a vote to the actual govDAO implementation.
+// If the voter cannot vote the specified proposal, this method will return an error
+// with the explanation of why.
+func VoteOnProposal(r VoteRequest) error {
+	return dao.VoteOnProposal(r)
+}
+
+func MustGetProposal(pid ProposalID) *Proposal {
+	p, err := GetProposal(pid)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return p
+}
+
+// GetProposal gets created proposal by its ID
+func GetProposal(pid ProposalID) (*Proposal, error) {
```

Comment:
Wouldn't a call to this function or `MustGetProposal()` allow overwriting `Proposal.Executor`? For example:

```go
MustGetProposal(2).Executor = nil
```

Reply:
Yes. I'll address this problem making it private like Callback.

Reply:
Moved the executor to a private field. Because of that, the method to execute proposals has to be moved to the proxy realm.